src/utils/mod.rs

```diff
@@ -63,6 +63,28 @@ pub(crate) fn expand_bits(bit_depth: u8, row_size: u32, buf: &[u8]) -> Vec<u8> {
     p
 }
 
+/// Checks if the provided dimensions would cause an overflow.
+/// 
+/// Returns the optionally provided `ImageError` on overflow or defaults to `LimitErrorKind::DimensionError`.
+#[allow(dead_code)]
+// When no image formats that use it are enabled
+pub(crate) fn check_dimension_overflow(
+    width: u32,
+    height: u32,
+    bytes_per_pixel: u8,
+    error: Option<crate::ImageError>,
+) -> crate::ImageResult<()> {
```

Comment:
This requires that we always pass an `error` even when the check succeeds. (Also, the `None` case isn't actually used?). A slight more efficient interface would remove those cost by taking a constructor instead:

```suggestion
pub(crate) fn check_dimension_overflow(
    width: u32,
    height: u32,
    bytes_per_pixel: u8,
    error: impl FnOnce() -> crate::ImageError,
) -> crate::ImageResult<()> {
```

Reply:
Yea, I initially thought I would have some cases were it would make sense to provide a default error. Since that is not the case I would probably take @fintelia's [suggestion](https://github.com/image-rs/image/pull/1407#discussion_r563168041) and return a bool. That seems a bit more flexible and obvious.